src/main/webapp/app/exercises/shared/rating/rating.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { RatingService } from 'app/exercises/shared/rating/rating.service';
+import { StarRatingComponent } from 'ng-starrating';
+import { Result } from 'app/entities/result.model';
+import { Rating } from 'app/entities/rating.model';
+
+@Component({
+    selector: 'jhi-rating',
+    templateUrl: './rating.component.html',
+    styleUrls: ['./rating.component.scss'],
+})
+export class RatingComponent implements OnInit {
+    public rating: Rating;
+    public disableRating = false;
+    @Input() result: Result;
+
+    constructor(public ratingService: RatingService) {}
+
+    ngOnInit(): void {
+        if (!this.result || !this.result.submission) {
+            return;
+        }
+        this.ratingService.getRating(this.result.id).subscribe((rating) => {
```

Comment:
this means we have one additional REST call, when navigating into the page that shows the rating / allows the user to interact with it.

Can you please do the following: 
- Only load the rating if it is not yet available (in the passed result) --> You might need to add rating as reference to result then
- Check if you can load the rating together with the result. (using the fetch mechanism in the repository)

Reply:
We use the same approach for complaints.
For those, we also do an extra network call to see if they are present.
I think we need to do it that way if we want to keep our design goal of this being independent and integratable by a single line.
Right now the Result does not know about ratings.

If we change this, we would need to update the Resources we well to include this for other exercise types.

Reply:
That's exactly my concern, we then have 3 request: result, complaint and rating.
We can keep this for now, but I would like to have 1 request here in the future